fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/advancement/CriterionRegistry.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.advancement;
+
+import net.minecraft.advancement.criterion.Criterion;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.mixin.object.builder.CriteriaAccessor;
+
+/**
+ * Allows registering advancement criteria for triggers.
+ *
+ * <p>A registered criterion (trigger) can be retrieved through
+ * {@link net.minecraft.advancement.criterion.Criteria#getById(Identifier)}.</p>
+ *
+ * @see net.minecraft.advancement.criterion.Criteria
+ */
+public final class CriterionRegistry {
+	/**
+	 * Registers a criterion for a trigger for advancements.
+	 *
+	 * @param criterion the criterion registered
+	 * @throws IllegalArgumentException if a criterion with the same {@link
+	 *                                  Criterion#getId() id} exists
+	 */
+	public static void register(Criterion<?> criterion) {
```

Comment:
Any chance we can change this to `<T extends Criterion<?>> T register(T criterion)` ? It would keep this method in line with the vanilla method, and allow inlined registration in a field initialization.

Reply:
could we get this added? I think it'd be good for merge after that @liach

Reply:
i personally don't like such call chaining

Reply:
...why?

Reply:
Personally I think the chaining is useful for people who have a giant `ModCriteria` class where they just want a static list of criteria.